src/Illuminate/Foundation/Testing/Constraints/HasSource.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Illuminate\Foundation\Testing\Constraints;
+
+class HasSource extends PageConstraint
+{
+    /**
+     * @var string
+     */
+    protected $source;
+
+    public function __construct($source)
+    {
+        $this->source = $source;
+    }
+
+    /**
+     * Check if the source is found in the given crawler.
+     *
+     * @param  \Symfony\Component\DomCrawler\Crawler  $crawler
+     * @return bool
+     */
+    protected function matches($crawler)
```

Comment:
similar to the above, I can't or it will raise a PHP error because of the same method defined in a parent class.